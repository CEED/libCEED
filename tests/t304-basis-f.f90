!-----------------------------------------------------------------------
      program test

      include 'ceedf.h'

      integer ceed,err
      real*8 a(16), lambda(4)

      character arg*32

      a = (/ 0.2, 0.0745355993, -0.0745355993, 0.0333333333,&
     &       0.0745355993, 1., 0.1666666667, -0.0745355993,&
     &      -0.0745355993, 0.1666666667, 1., 0.0745355993,&
     &      0.0333333333, -0.0745355993, 0.0745355993, 0.2 /)

      call getarg(1,arg)

      call ceedinit(trim(arg)//char(0),ceed,err)
      call ceedsymmetricschurdecomposition(ceed,a,lambda,4,err);
      write (*,*) 'Q:'
      do i=0,3
        do j=1,4
          if (abs(a(i*4+j))<1.0D-14) then
! LCOV_EXCL_START
            a(i*4+j) = 0
! LCOV_EXCL_STOP
          endif
        enddo
        write(*,'(A,F12.8,F12.8,F12.8,F12.8)') '',&
     &   a(i*4+1),a(i*4+2),a(i*4+3),a(i*4+4)
      enddo
      write (*,*) 'lambda:'
      do i=1,4
        if (abs(lambda(i))<1.0D-14) then
! LCOV_EXCL_START
          lambda(i) = 0
! LCOV_EXCL_STOP
        endif
        write(*,'(A,F12.8)') '',lambda(i)
      enddo

      call ceeddestroy(ceed,err)

      end
!-----------------------------------------------------------------------
