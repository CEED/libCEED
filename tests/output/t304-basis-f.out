 Q:
  0.14758817 -0.70710678 -0.69153289  0.00000000
  0.69153289  0.00000000  0.14758817 -0.70710678
 -0.69153289  0.00000000 -0.14758817 -0.70710678
 -0.14758817 -0.70710678  0.69153289  0.00000000
 lambda:
  0.86514837
  0.23333334
  0.13485163
  1.16666667
